web/src/main/java/com/bazaarvoice/emodb/web/scanner/writer/KafkaScanDestinationWriter.java

```diff
@@ -0,0 +1,156 @@
+package com.bazaarvoice.emodb.web.scanner.writer;
+
+import com.bazaarvoice.emodb.sor.api.Coordinate;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Optional;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class writes documents to Kafka asynchronously, then cleans up the futures using the future service. Access to
+ * this class is multi-threaded. A single thread is responsible for calling {@link #writeDocument(Map)},
+ * {@link #closeAndCancel()}, and {@link #closeAndTransferAsync(Optional)}. A separate threads calls {@link #isFinishedUploading()}
+ */
+public class KafkaScanDestinationWriter implements ScanDestinationWriter {
+
+    private static Logger _log = LoggerFactory.getLogger(KafkaScanDestinationWriter.class);
+
+    private final Producer<String, JsonNode> _producer;
+    private final ObjectMapper _mapper;
+    private final String _topic;
+    private final BlockingQueue<CoordinateAndFuture> _futureQueue;
+    private final ExecutorService _futureGettingService;
+    private final AtomicReference<Throwable> _error = new AtomicReference<>();
+    private volatile boolean _closed;
+    private int _bytesTransferred;
+    private int _bytesAdded;
+
+    private final Meter _blockingQueueFullMeter;
+
+
+    public KafkaScanDestinationWriter(Producer<String, JsonNode> producer, ObjectMapper objectMapper, String topic,
+                                      MetricRegistry metricRegistry) {
+        _producer = producer;
+        _mapper = objectMapper;
+        _topic = topic;
+        _futureQueue = new ArrayBlockingQueue<>(10000);
```

Comment:
out of curiosity, why it's set to `10000`?

Reply:
The goal was to set this to a number that was to set this a number high enough that the BlockingQueue would not fill up, but no so high that we threaten memory consumption.

Reply:
I am going to extract it to a constant.